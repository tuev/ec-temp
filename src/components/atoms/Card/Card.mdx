# Stories mdx for Card

import { Meta, Story } from '@storybook/addon-docs/blocks'
import * as stories from './Card.stories.tsx'
import Card from './Card.tsx'

<Story name="basic" />