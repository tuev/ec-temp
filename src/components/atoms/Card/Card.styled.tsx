/**
 *
 * Styled for Card
 *
 */

import styled from 'styled-components'

export const CardWrapper = styled.div`
  cursor: pointer;
  will-change: transform, opacity;
  box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.2);
  min-height: 335px;
  padding: 8px;

  &:hover {
    box-shadow: 0 20px 40px 0 rgba(0, 0, 0, 0.3);
  }
`
