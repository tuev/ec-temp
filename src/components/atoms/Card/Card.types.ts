export interface ICardProps {
  children?: React.ReactChildren | string | React.ReactNode
}

export type CardProps = ICardProps
