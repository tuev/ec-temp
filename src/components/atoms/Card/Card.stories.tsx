/**
 *
 *  Stories for Card
 *
 */

import React from 'react'
import { action } from '@storybook/addon-actions'
import Card from './Card'
import { Grid } from '@material-ui/core'

export default {
  component: Card,
  includeStories: ['basic'],
  title: 'Card',
}

export const basic = () => (
  <Grid container={true} spacing={2}>
    <Grid item={true} xs={3}>
      <Card>Item</Card>
    </Grid>
    <Grid item={true} xs={3}>
      <Card>Item</Card>
    </Grid>
    <Grid item={true} xs={3}>
      <Card>Item</Card>
    </Grid>
    <Grid item={true} xs={3}>
      <Card>Item</Card>
    </Grid>
  </Grid>
)
