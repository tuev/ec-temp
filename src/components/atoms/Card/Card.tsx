/**
 *
 * Card
 *
 */

import React, { memo, FC } from 'react'
import { CardWrapper } from './Card.styled'
import { CardProps } from './Card.types'

const Card: FC<CardProps> = ({ children, ...rest }) => {
  return <CardWrapper {...rest}>{children}</CardWrapper>
}

export default memo(Card)
