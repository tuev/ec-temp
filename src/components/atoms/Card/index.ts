/**
 *
 * Index for Card
 *
 */

import Card from './Card'

export default Card
