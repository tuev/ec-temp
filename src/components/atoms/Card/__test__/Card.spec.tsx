/**
 *
 * Tests for Card
 *
 */

import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import Card from '..'

describe('App Card', () => {
  it('Test get default Card no params ', () => {
    // test snapshot
    const wrapper = render(<Card />)
    expect(wrapper).toMatchSnapshot()
  })
})
