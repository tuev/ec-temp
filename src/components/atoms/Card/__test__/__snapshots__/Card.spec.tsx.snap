// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Card Test get default Card no params  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c0 {
  cursor: pointer;
  will-change: transform,opacity;
  box-shadow: 0 1px 1px 0 rgba(0,0,0,0.2);
  min-height: 335px;
  padding: 8px;
}

.c0:hover {
  box-shadow: 0 10px 20px 0 rgba(0,0,0,0.3);
}

<body>
    <div>
      <div
        class="c0"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxjAm jewvia"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
